codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/SmithyGoDependency.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+/**
+ * An enum of all the built-in dependencies used by this package.
+ */
+public enum SmithyGoDependency {
+
+    // The version in the stdlib dependencies should reflect the minimum Go version.
+    // The values aren't currently used, but they could potentially used to dynamically
+    // set the minimum go version.
+    BIG(stdlib("math/big")),
+    TIME(stdlib("time")),
+    FMT(stdlib("fmt")),
+    CONTEXT(stdlib("context")),
```

Comment:
Would it make sense for these to be public static final members of `SmithyGoDependency` as type `GoDependency` and not enums? If they were that style they could still be treated as if they were enums in our code, and the extra `getDependency` method call wouldn't be needed.

```java
public class SmithyGoDependency {
      public static final GoDependency BIG = stdlib("math/big") // or what ever the equivalent is.
...
}
```